prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusStorageFactory.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.sql.prometheus.storage;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import okhttp3.OkHttpClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.sql.catalog.model.ConnectorType;
+import org.opensearch.sql.catalog.model.auth.AuthenticationType;
+import org.opensearch.sql.prometheus.authinterceptors.AwsSigningInterceptor;
+import org.opensearch.sql.prometheus.authinterceptors.BasicAuthenticationInterceptor;
+import org.opensearch.sql.prometheus.client.PrometheusClient;
+import org.opensearch.sql.prometheus.client.PrometheusClientImpl;
+import org.opensearch.sql.storage.StorageEngine;
+import org.opensearch.sql.storage.StorageEngineFactory;
+
+public class PrometheusStorageFactory implements StorageEngineFactory {
+
+  private static final Logger LOG = LogManager.getLogger();
+  public static final String URI = "prometheus.uri";
+  public static final String AUTH_TYPE = "prometheus.auth.type";
+  public static final String USERNAME = "prometheus.auth.username";
+  public static final String PASSWORD = "prometheus.auth.password";
+  public static final String REGION = "prometheus.auth.region";
+  public static final String ACCESS_KEY = "prometheus.auth.access_key";
+  public static final String SECRET_KEY = "prometheus.auth.secret_key";
```

Comment:
Are these constants used elsewhere? It's fine if not. Otherwise, I think we may move them to some constant class.